src/systemctl/systemctl.c

```diff
@@ -165,12 +166,18 @@ static bool arg_jobs_before = false;
 static bool arg_jobs_after = false;
 static char **arg_clean_what = NULL;
 
+/* This is a global cache that will be constructed on first use. */
+Hashmap *arg_id_map = NULL;
+Hashmap *arg_name_map = NULL;
```

Comment:
why not static?

Reply:
why prefixed with arg_? This isn't precisely something that is derived directly from the command line or an env var or so?

Reply:
We had a general rule that global args are prefixed with `arg_`, no? I think you said so yourself.

Reply:
Well, we had a rule of avoiding them, as much as we can except if they inherently are a per-process concept, i.e. something derived directly from the process' comand line or environment block... Those two are summarized as "arguments", and hence get the "arg_" prefix... However, all other global variables should get something else, and be avoided unless we have a really good reason to have them...

https://systemd.io/CODING_STYLE#command-line
https://systemd.io/CODING_STYLE#code-organization-and-semantics